src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs

```diff
@@ -330,46 +331,61 @@ public async Task GetAll()
 
                 await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
                 {
-                    writer.WriteStartObject();
+                    WriteGetAllResult(writer, subscriptions, context);
+                }
+            }
+        }
 
-                    var subscriptionsAsBlittable = subscriptions.Select(x => new DynamicJsonValue()
-                    {
-                        [nameof(SubscriptionState.SubscriptionId)] = x.SubscriptionId,
-                        [nameof(SubscriptionState.SubscriptionName)] = x.SubscriptionName,
-                        [nameof(SubscriptionState.ChangeVectorForNextBatchStartingPoint)] = x.ChangeVectorForNextBatchStartingPoint,
-                        [nameof(SubscriptionState.Query)] = x.Query,
-                        [nameof(SubscriptionState.Disabled)] = x.Disabled,
-                        [nameof(SubscriptionState.LastClientConnectionTime)] = x.LastClientConnectionTime,
-                        [nameof(SubscriptionState.LastBatchAckTime)] = x.LastBatchAckTime,
-                        ["Connection"] = GetSubscriptionConnectionJson(x.Connection),
-                        ["Connections"] = GetSubscriptionConnectionsJson(x.Connections),
-                        ["RecentConnections"] = x.RecentConnections?.Select(r => new DynamicJsonValue()
-                        {
-                            ["State"] = new DynamicJsonValue()
-                            {
-                                ["LatestChangeVectorClientACKnowledged"] = r.SubscriptionState.ChangeVectorForNextBatchStartingPoint,
-                                ["Query"] = r.SubscriptionState.Query
-                            },
-                            ["Connection"] = GetSubscriptionConnectionJson(r)
-                        }),
-                        ["FailedConnections"] = x.RecentRejectedConnections?.Select(r => new DynamicJsonValue()
-                        {
-                            ["State"] = new DynamicJsonValue()
-                            {
-                                ["LatestChangeVectorClientACKnowledged"] = r.SubscriptionState.ChangeVectorForNextBatchStartingPoint,
-                                ["Query"] = r.SubscriptionState.Query
-                            },
-                            ["Connection"] = GetSubscriptionConnectionJson(r)
-                        }).ToList()
-                    });
+        internal static void WriteGetAllResult(AsyncBlittableJsonTextWriter writer, IEnumerable<SubscriptionState> subscriptions, TransactionOperationContext context)
```

Comment:
those things needs to be cleaned as soon as processors will be implemented

Reply:
mentioned in https://issues.hibernatingrhinos.com/issue/RavenDB-18104